ports/servoshell/desktop/minibrowser.rs

```diff
@@ -182,75 +184,69 @@ impl Minibrowser {
-
-        // Expansion would also show that they are 2 separate widgets
-        visuals.widgets.active.expansion = 0.0;
-        visuals.widgets.hovered.expansion = 0.0;
-        // The rounding is changed so it looks as though the 2 widgets are a single widget
-        // with a uniform rounding
-        let corner_radius = egui::CornerRadius {
-            ne: 0,
-            nw: 4,
-            sw: 4,
-            se: 0,
-        };
-        visuals.widgets.active.corner_radius = corner_radius;
-        visuals.widgets.hovered.corner_radius = corner_radius;
-        visuals.widgets.inactive.corner_radius = corner_radius;
-
+        let inactive_bg_color = ui.visuals().window_fill;
+        let active_bg_color = ui.visuals().widgets.active.weak_bg_fill;
         let selected = webview.focused();
-        let tab = ui.add(SelectableLabel::new(
-            selected,
-            truncate_with_ellipsis(&label, 20),
-        ));
-        let tab = tab.on_hover_ui(|ui| {
-            ui.label(label);
-        });
 
-        let corner_radius = egui::CornerRadius {
-            ne: 4,
-            nw: 0,
-            sw: 0,
-            se: 4,
-        };
-        let visuals = ui.visuals_mut();
-        visuals.widgets.active.corner_radius = corner_radius;
-        visuals.widgets.hovered.corner_radius = corner_radius;
-        visuals.widgets.inactive.corner_radius = corner_radius;
+        // Setup a tab frame that will contain the favicon, title and close button
+        let mut tab_frame = egui::Frame::NONE.corner_radius(4).begin(ui);
+        {
+            tab_frame.content_ui.add_space(5.0);
+
+            let visuals = tab_frame.content_ui.visuals_mut();
+            // Remove the stroke so we don't see the border between the close button and the label
+            visuals.widgets.active.bg_stroke.width = 0.0;
+            visuals.widgets.hovered.bg_stroke.width = 0.0;
+            // Now we make sure the fill color is always the same, irrespective of state, that way
+            // we can make sure that both the label and close button have the same background color
+            visuals.widgets.noninteractive.weak_bg_fill = inactive_bg_color;
+            visuals.widgets.inactive.weak_bg_fill = inactive_bg_color;
+            visuals.widgets.hovered.weak_bg_fill = active_bg_color;
+            visuals.widgets.active.weak_bg_fill = active_bg_color;
+            visuals.selection.bg_fill = active_bg_color;
+            visuals.selection.stroke.color = visuals.widgets.active.fg_stroke.color;
+            visuals.widgets.hovered.fg_stroke.color = visuals.widgets.active.fg_stroke.color;
+
+            // Expansion would also show that they are 2 separate widgets
+            visuals.widgets.active.expansion = 0.0;
+            visuals.widgets.hovered.expansion = 0.0;
+
+            if let Some(favicon_url) = webview.favicon_url() {
+                tab_frame.content_ui.add(
+                    egui::Image::new(favicon_url.to_string())
```

Comment:
What does this do under the hood? Does egui have a networking stack? I'm ok with that in the short term, but we should make it possible for the engine to fetch the icon and the embedder to use the cached response body instead.

Reply:
Yeah, the plan is that eventually the `WebView` API should expose the favicon as a bitmap as it should be fetched using the cookies, etc of the context that it is loading in.

Reply:
> Does egui have a networking stack?

Yes, with `egui_extras`. I also don't like this solution, but it's kinda the only possibility given the current embedding API